pkg/parser/custom/py/setuppy_parser.go

```diff
@@ -0,0 +1,233 @@
+				child := node.Child(int(i))
+				a, b := s.extractStringConstants(child, code)
+				const_strings2 = append(const_strings2, a...)
+				symbol_strings2 = append(symbol_strings2, b...)
+			}
+			const_strings = append(const_strings, const_strings2...)
+			symbol_string = append(symbol_string, symbol_strings2...)
+
+			s.Symbol2strings[attribute.Content(code)] = const_strings2
+			s.Symbol2symbols[attribute.Content(code)] = symbol_strings2
+		}
+	}
+	case "call":
+		// fmt.Println(node.ChildCount() , node.Child(0).Type(), node.Child(0).Content(code))
+		for i := uint32(0); i < node.ChildCount(); i++ {
+			child := node.Child(int(i))
+			a, b := s.extractStringConstants(child, code)
+			const_strings = append(const_strings, a...)
+			symbol_string = append(symbol_string, b...)
+		}
+	default:
+		for i := uint32(0); i < node.ChildCount(); i++ {
+			child := node.Child(int(i))
+			a, b := s.extractStringConstants(child, code)
+			const_strings = append(const_strings, a...)
+			symbol_string = append(symbol_string, b...)
+		}
+	}
+
+	return const_strings, symbol_string
+}
+
+// aggStringConstants aggregates string constants based on dependencies.
+func (s *SetuppyParserViaSyntaxTree) aggStringConstants(src string) []string {
+	const_strings := []string{}
+
+	if cs, ok := s.Symbol2strings[src]; ok {
+		const_strings = append(const_strings, cs...)
+	} 
+
+	var dep_syms []string
+	if ds, ok := s.Symbol2symbols[src]; !ok {
+		return const_strings
+	} else {
+		dep_syms = ds
+	}
+
+	for _, sym := range dep_syms {
+		a := s.aggStringConstants(sym)
+		const_strings = append(const_strings, a...)
+	}
+
+	//Remove " char
+
+
+	//Remvove Duplicates
+	// unique.Sort(unique.StringSlice{&const_strings})
+	// fmt.Printf("%s", strings.Join(const_strings[:], ","))
+
+	return const_strings
+}
+
+/**
+ GetDependencyStrings extracts dependency strings from a setup.py file.
+
+ - "iptools>=0.7.0"
+- "parsedatetime>=2.4"
+- "beautifulsoup4>=4.7.1"
+- "fuzzywuzzy>=0.18.0"
+- "PySocks>=1.7.0"
+- "truffleHogRegexes>=0.0.7"
+- "soupsieve>=1.9.1"
+- "filetype>=1.0.5"
+- "pyunpack>=0.1.2"
+- "patool>=1.12"
+- "wordninja>=2.0.0"
+- "iocextract>=1.13.1"
+- "pyparsing>=3.0.8"
+- "ioc-fanger"
+- "titlecase>=0.12.0"
+- "furl>=2.1.0"
+- "pathlib2>=2.3.3"
+- "lxml>=4.5.0"
+- "food-exceptions>=0.4.4"
+- "food-models>=3.3.1"
+- "dateutils>=0.6.6"
+- "publicsuffixlist>=0.6.2"
+- "dnspython"
+- "netaddr>=0.7.18"
+- "validators>=0.12.2"
+- "fqdn>=1.1.0"
+- "tld>=0.9.1"
+- "cchardet>=2.1.4"
+- "urllib3>=1.22"
+- "tldextract>=2.2.0"
+*/
+func (s *SetuppyParserViaSyntaxTree) GetDependencyStrings(filepath string) ([]string, error) {
+
+	var dependencies []string
+	var code []byte
+
+	if content, err := os.ReadFile(filepath); err != nil {
+		logger.Warnf("Error opening setuppy file %v", err)
+		return dependencies, err
+	} else {
+		code = content
+	}
+
+	ctx := context.Background()
+	tree, err := s.Parser.ParseCtx(ctx, nil, code)
+
+	if err != nil{
```

Comment:
Potential formatting error. Can you run `go fmt` or install Go language support in IDE?